src-java/rule-manager/rule-manager-api/src/main/java/org/openkilda/rulemanager/Instructions.java

```diff
@@ -0,0 +1,33 @@
+/* Copyright 2021 Telstra Open Source
+ *
+ *   Licensed under the Apache License, Version 2.0 (the "License");
+ *   you may not use this file except in compliance with the License.
+ *   You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   Unless required by applicable law or agreed to in writing, software
+ *   distributed under the License is distributed on an "AS IS" BASIS,
+ *   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *   See the License for the specific language governing permissions and
+ *   limitations under the License.
+ */
+
+package org.openkilda.rulemanager;
+
+import org.openkilda.rulemanager.action.Action;
+
+import com.fasterxml.jackson.databind.annotation.JsonSerialize;
+import lombok.Builder;
+
+import java.util.List;
+
+@JsonSerialize
+@Builder
+public class Instructions {
+
+    List<Action> applyActions;
+    List<Action> setActions;
```

Comment:
I still propose to use `Set` for data represens some set of records. We must "tell" that there is no way to define other or to use same action type more that one time.

Reply:
Changed.